xarray/coding/times.py

```diff
@@ -363,24 +437,36 @@ def decode_cf_datetime(
 
 
 def to_timedelta_unboxed(value, **kwargs):
+    # todo: check, if the procedure here is correct
     result = pd.to_timedelta(value, **kwargs).to_numpy()
-    assert result.dtype == "timedelta64[ns]"
+    unique_timedeltas = np.unique(result[pd.notnull(result)])
+    unit = _netcdf_to_numpy_timeunit(_infer_time_units_from_diff(unique_timedeltas))
+    if unit not in ["s", "ms", "us", "ns"]:
+        unit = "s"
```

Comment:
Should this raise an error instead?

Reply:
I think no. `pd.to_timedelta().to_numpy()` returns `timedelta64[ns]`. The following check and astype just make sure, that the result is in the lowest possible resolution but within the range of `TimedeltaIndex` (["s", "ms", "us", "ns"]).